internal/controller/progressive/progressive.go

```diff
@@ -520,16 +532,16 @@ func startUpgradeProcess(
 }
 
 // return true if all Conditions are true
-func checkChildConditions(upgradingObjectStatus *kubernetes.GenericStatus) bool {
+func checkChildConditions(upgradingObjectStatus *kubernetes.GenericStatus) (bool, string) {
```

Comment:
from this function instead of returning a string, can we return the first Condition that failed?

Then, on the caller side create the string which includes both the name of the Condition, the fact that it was false, and the `c.Reason`.